transmart-rest-api/src/main/groovy/org/transmartproject/rest/dataExport/ExportJobExecutor.groovy

```diff
@@ -0,0 +1,99 @@
+package org.transmartproject.rest.dataExport
+
+import grails.persistence.support.PersistenceContextInterceptor
+import grails.util.Holders
+
+import groovy.util.logging.Slf4j
+import org.apache.commons.lang.StringUtils
+import org.quartz.Job
+import org.quartz.JobExecutionContext
+import org.transmartproject.core.exceptions.UnexpectedResultException
+
+import java.util.zip.ZipOutputStream
+
+/**
+ * This class will encompass the job scheduled by Quartz.
+ */
+@Slf4j
+class ExportJobExecutor implements Job {
+
+    def ctx = Holders.grailsApplication.mainContext
+
+    RestDataExportService restDataExportService = ctx.restDataExportService
+    ExportAsyncJobService asyncJobService = ctx.exportAsyncJobService
+//    QuartzSpringScope quartzSpringScope = ctx.quartzSpringScope
+
+    final String tempFolderDirectory = Holders.config.com.recomdata.plugins.tempFolderDirectory
+
+    public void execute(JobExecutionContext jobExecutionContext) {
+        def userInContext = jobExecutionContext.jobDetail.jobDataMap['userInContext']
+        Map jobDataMap = jobExecutionContext.jobDetail.getJobDataMap()
+
+//        // put the user in context
+//        quartzSpringScope.currentUserBeanQuartzScope = userInContext
```

Comment:
If the user is to be injected in the background job something similar will be needed. But I don't know if the user needs to be injected and i would say preferrably not (though I don't know what the spring best practices would be in such a context).